tiphub-io/tiphub
# Security & Performance Optimizations

No tracked issue.

### What This Does

#### Security Improvements

* Enforces HTTPS in production
* Adds a content security policy that avoids loading malicious content
* Removes all external asset dependencies
* Adds CORS_DOMAINS environment variable for configuration to prevent CSRF
* Other misc. improvements from flask-talisman

#### Performance Optimizations

* Removes inline source maps from production js, uses external map files instead
* Optimized the moment and react-syntax-highlighter bundles to not include all locales / languages by default, only the ones we use
* CSS properly minified
* gzip all static content for massive savings

### Steps to Test

* Run development version, confirm CSP is in place
* Run production version, confirm you are forced onto HTTPS (This will not work in development as you won't be serving up with a cert)
* Run production version but disable enforced HTTPS (app.py:31) and check out the sick bundle size savings

## Screenshots

### Before

<img width="323" alt="Screen Shot 2019-05-08 at 4 21 16 PM" src="https://user-images.githubusercontent.com/649992/57405880-84096580-71ad-11e9-980e-ce5706135aa1.png">


### After

<img width="324" alt="Screen Shot 2019-05-08 at 4 21 00 PM" src="https://user-images.githubusercontent.com/649992/57405883-87045600-71ad-11e9-9ad5-c41239f1f0a7.png">
